### DISCUSSION TITLE
Docker images publication workflow

### DISCUSSION THREAD

**Discussion by Tansito** [Qiskit/qiskit-serverless#764] [Ideas]
Related with #690 I was having some discussions with the dev-ops teams related with our publication workflow. Right now what we have is the next one:

- `latest`: tag for `main` branch
- `x.y.z`: semver tag for `releases`

This makes that we need to run several processes in all the Pull Requests because all the changes that we merge in `main` are published. The workflow proposed here is the next one:

- `x.y.z`: semver tag for `releases`
- `latest`: tag pointing out to the last `release`

No tag `initially` for `main` but there are tags for those use cases like `nightly`. I think we can skip it because the current use of `main` is near to zero. And the benefits that we would receive with this change is:

- Some resources optimization: there are steps in the CI that we could left only for releases and simplify the pull requests.
- Reduce the number of pushes to the registry: they are aware of these numbers, there is no problem right now but just to be aware with.
- More near to the standard: it seems the `latest` tag is typically used for that purpose. A little near example is [ray-image](https://hub.docker.com/r/rayproject/ray).

I liked the idea and I think it has sense but I would like to know yours!

**Comment by psschwei** [comment]
I'm totally fine with dropping a build for every commit (and it's easy enough to do locally if one really needs it). 

I'd also be ok with dropping the `latest` tag in general (it's not good practice to use it), but if it's pointing to the last release, then :shrug: 

We could also think about using digests instead of tags, but that's probably a discussion for another day :smile:
